src/context/filters/filtersSlice.ts

```diff
@@ -23,20 +25,30 @@ export const slice = createSlice({
       ...state,
       layers: layers.add(payload),
     }),
+
     removeLayer: (
       { layers, ...state },
       { payload }: PayloadAction<LayerType>,
     ) => ({
       ...state,
       layers: layers.delete(payload),
     }),
+
+    updateDate: ({ layers, ...state }, { payload }: PayloadAction<Date>) => ({
+      ...state,
+      // 'layers' is getting converted into a object instead of Set<LayerType>
+      layers: layers as Set<LayerType>,
+      currentDate: payload,
+    }),
   },
 });
 
 // Getters
-export const selectlayers = (state: RootState) => state.filters.layers;
+export const selectLayers = (state: RootState) => state.filters.layers;
+export const selectCurrentDate = (state: RootState) =>
```

Comment:
If you keep immutable state you _probably_ can still use destructoring on the selectors if you want.